Find rpcgen using the name with which it's installed by the MacPorts rpcgen-mt port.
--- meson.build.orig	2023-10-02 02:31:09.000000000 -0500
+++ meson.build	2023-10-09 19:30:47.000000000 -0500
@@ -801,7 +801,7 @@
 ]
 
 required_programs_groups = [
-  { 'name': 'rpcgen', 'prog': [ 'rpcgen', 'portable-rpcgen' ] },
+  { 'name': 'rpcgen', 'prog': [ 'rpcgen-mt', 'rpcgen', 'portable-rpcgen' ] },
 ]
 
 if host_machine.system() == 'freebsd'
